spec/forms/steps/case/charges_form_spec.rb

```diff
@@ -28,4 +28,38 @@
       end
     end
   end
+
+  describe '#offence_dates' do
+    it 'returns a list of offence date fieldset forms' do
+      charge_record.offence_dates << [
+        OffenceDate.new(date: "02, 02, 2000"),
+        OffenceDate.new(date: "03, 10, 1998"),
+        OffenceDate.new(date: "11, 12, 2021")
+      ]
+
+      subject.offence_dates.each do |form|
```

Comment:
Maybe shorter:
`expect(subject.offence_dates).to all(be_a(Steps::Case::OffenceDateFieldsetForm))`